pkg/controller/cassandradatacenter/controller/configmap.go

```diff
@@ -157,14 +121,90 @@ func addCassandraYamlOverrides(config configurationResources) error {
 	})
 
 	if err != nil {
-		return err
+		// we're serializing a known structure to YAML -- if that fails...
+		panic(err)
 	}
 
-	configMapVolumeAddTextFile(config.configMap, config.volumeSource, "cassandra.yaml.d/001-operator-overrides.yaml", string(data))
+	addFileFn("cassandra.yaml.d/001-operator-overrides.yaml", string(data))
+}
+
+
+
+func addCassandraJVMOptions(cdc *cassandraoperatorv1alpha1.CassandraDataCenter, addFileFn func (path string, data string)) {
```

Comment:
Moved from `jvm.go`, which seemed like an arbitrary split between files -- everything config related is here, except for JVM config?